Nodejs/Product/Nodejs/Options/NodejsNpmOptionsControl.cs

```diff
@@ -42,20 +36,26 @@ public partial class NodejsNpmOptionsControl : UserControl {
         }
 
         private void ClearCacheButton_Click(object sender, EventArgs e) {
+            bool didClearNpmCache = DeleteCacheDirectory("npm cache", NodejsConstants.NpmCachePath);
+            bool didClearTools = DeleteCacheDirectory("NTVS external tools", NodejsConstants.ExternalToolsPath);
+
+            _cacheClearedSuccessfully.Visible = (didClearNpmCache && didClearTools);
+        }
+
+        private static bool DeleteCacheDirectory(string displayName, string cachePath) {
             try {
-                Directory.Delete(_npmCachePath, true);                    
-                _cacheClearedSuccessfully.Visible = true;
+                Directory.Delete(cachePath, true);
+               return true;
             } catch (DirectoryNotFoundException) {
                 // Directory has already been deleted. Do nothing.
-                _cacheClearedSuccessfully.Visible = true;
+                return true;
             } catch (IOException exception) {
                 // files are in use or path is too long
                 MessageBox.Show(
-                           string.Format("Cannot clear npm cache. {0}", exception.Message),
-                           "Cannot Clear npm Cache",
-                           MessageBoxButtons.OK,
-                           MessageBoxIcon.Information
-                       );
+                    string.Format("Cannot clear {0}. {1}", displayName, exception.Message),
+                    string.Format("Cannot Clear {0}", displayName),
+                    MessageBoxButtons.OK,
+                    MessageBoxIcon.Information);
```

Comment:
If neither are able do be deleted successfully, we're going to display two message boxes in a row, which is a pretty poor user experience. Maybe we should just point to the Node.js Tools directory as a whole?